src/main/java/com/znsio/rpi/properties/ReportPortalPropertiesOverloader.java

```diff
@@ -0,0 +1,169 @@
+        addAttributes("OS", System.getProperty("os.name"));
+        addAttributes("Username", System.getProperty("user.name"));
+        try {
+            if (Strings.isNotNullAndNotEmpty(InetAddress.getLocalHost().getHostName())) {
+                String hostName = InetAddress.getLocalHost().getHostName();
+                addAttributes("HostName", hostName);
+            }
+        } catch (UnknownHostException ex) {
+            LOGGER.warn("Unable to set Report Portal " +
+                    "Attributes for HostName: " + ex);
+        }
+    }
+
+    private static void setTestAttributes() {
+        addAttributes("TargetEnvironment", config.getProperty(Config.TARGET_ENVIRONMENT));
+        addAttributes("Platform", config.getProperty(Config.PLATFORM).toUpperCase());
+        if (isPlatformWeb()) {
+            addAttributes("Browser", config.getProperty(Config.BROWSER));
+        } else {
+            addAttributes("App", config.getProperty(Config.APP_PACKAGE_NAME));
+            addAttributes("LocalDeviceExecution", config.getProperty(Config.IS_LOCAL_DEVICE));
+        }
+        addAttributes("VisualEnabled", getOverriddenStringValue(Config.IS_VISUAL,
+                config.getProperty(Config.IS_VISUAL, "false")));
+        addAttributes("ParallelCount", Integer.toString(getThreadCount()));
+    }
+
+    private static void setPipelineAttributes() {
+        if (Boolean.parseBoolean(getOverriddenStringValue(Config.RUN_IN_CI, config.getProperty(Config.RUN_IN_CI)))) {
+            addAttributes("RunInCI", "true");
+            addAttributes("PipelineExecutionID", getOverriddenStringValue(Config.BUILD_ID,
+                    getOverriddenStringValue(config.getProperty(Config.BUILD_ID), NOT_SET)));
+            addAttributes("AgentName", getOverriddenStringValue(Config.AGENT_NAME,
+                    getOverriddenStringValue(config.getProperty(Config.AGENT_NAME), NOT_SET)));
+            addAttributes("AutomationBranch", getOverriddenStringValue(Config.BRANCH_NAME,
+                    getOverriddenStringValue(config.getProperty(Config.BRANCH_NAME), getBranchNameUsingGitCommand())));
+        } else {
+            addAttributes("RunInCI", "false");
+            addAttributes("AutomationBranch", getBranchNameUsingGitCommand());
+        }
+    }
+
+    private static String getBranchNameUsingGitCommand() {
+        String[] getBranchNameCommand = new String[]{"git", "rev-parse", "--abbrev-ref", "HEAD"};
+        CommandLineResponse response = CommandLineExecutor.execCommand(getBranchNameCommand);
+        String branchName = response.getStdOut();
+        LOGGER.info(String.format("\tBranch name from git command: '%s': '%s'",
+                Arrays.toString(getBranchNameCommand), branchName));
+        return branchName;
+    }
+
+    private static String getBrowserVersion(String browserPath) {
+        String[] getBrowserVersionCommand = new String[]{browserPath, "-v", "|", "awk", "'{print $2}'"};
```

Comment:
will this work on Windows?

Reply:
No, it won't. I don't think there's straight forward solution for windows. The `addAttributes` method will not add any attribute to the RP if either the key or the valye is null or empty.

Reply:
Added Todo as discussed